src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -104,7 +104,7 @@ export class HttpConnection implements IConnection {
             }
         }
 
-        this.httpClient = options.httpClient || new DefaultHttpClient(this.logger);
+        this.httpClient = options.httpClient || new DefaultHttpClient(this.logger, options.headers);
```

Comment:
Unfortunately this doesn't allow using headers from options with a custom httpclient.

I think we would want a `setHeaders()` function on `HttpClient`.

Reply:
https://github.com/dotnet/aspnetcore/pull/19227/commits/cc8eaa48f3b4d13e2488ae63f6d37a9b8cf871cc